src/Processors/QueryPlan/RuntimeFilterLookup.cpp

```diff
@@ -112,41 +112,41 @@ void RuntimeFilter::addAllFrom(const RuntimeFilter & source)
     }
 }
 
-ColumnPtr RuntimeFilter::find(ColumnPtr values) const
+ColumnPtr RuntimeFilter::find(const ColumnWithTypeAndName & values) const
 {
     if (!inserts_are_finished)
         throw Exception(ErrorCodes::LOGICAL_ERROR, "Trying to lookup values in runtime filter before builiding it was finished");
 
     if (no_elements_in_set)
     {
         /// If the set is empty just return Const False column
-        return result_type->createColumnConst(values->size(), false);
+        return result_type->createColumnConst(values.column->size(), false);
     }
     else if (single_element_in_set)
     {
         /// If only 1 element in the set then use "value == const" instead of set lookup
-        auto const_column = filter_column_target_type->createColumnConst(values->size(), *single_element_in_set);
+        auto const_column = filter_column_target_type->createColumnConst(values.column->size(), *single_element_in_set);
         ColumnsWithTypeAndName equals_args = {
-            ColumnWithTypeAndName(values, filter_column_target_type, String()),
+            values,
```

Comment:
The fix is to pass original values column data type instead of saying that is has filter_column_target_type while it can be LowCardinality